packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -24,21 +25,27 @@ export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryRespon
     return result
 }
 
-export const useAsyncExecute = <A, R>(fn: (arg: A) => Promise<R>) => {
+export const useAsyncExecute = <A extends any[], R>(fn: (...args: A) => Promise<R>) => {
     // This is a stub implementation to validate the types.
     // The real implementation will be more React-y.
+    const [isLoading, setIsLoading] = useState<boolean>(false)
```

Comment:
```suggestion
    const [isLoading, setIsLoading] = useState(false)
```
TypeScript infers `boolean` from `false`, so no need to make it explicit.